storagenode/orders/service.go

```diff
@@ -74,99 +74,132 @@ type DB interface {
 	Archive(ctx context.Context, requests ...ArchiveRequest) error
 	// ListArchived returns orders that have been sent.
 	ListArchived(ctx context.Context, limit int) ([]*ArchivedInfo, error)
-	// CleanArchive deletes all entries older than ttl
-	CleanArchive(ctx context.Context, ttl time.Duration) (int, error)
+	// CleanArchive deletes all entries older than interval
+	CleanArchive(ctx context.Context, interval time.Duration) (int, error)
 }
 
-// SenderConfig defines configuration for sending orders.
-type SenderConfig struct {
-	Interval time.Duration `help:"duration between sending" default:"1h0m0s"`
-	Timeout  time.Duration `help:"timeout for sending" default:"1h0m0s"`
+// Config defines configuration for sending orders.
+type Config struct {
+	SenderInterval  time.Duration `help:"duration between sending" default:"1h0m0s"`
+	CleanupInterval time.Duration `help:"duration between archive cleanups. Each cleanup deletes all orders archived before the previous iteration, leaving the last period available for reference" default:"744h0m0s"`
```

Comment:
After looking at the code, I do think we should add another config interval, that triggers how often we should run the cleanup. I would name them something like this:

`CleanupInterval` <-- this controls how _often_ clean-up is triggered
`CleanupTTL` <-- this controls which orders are deleted

Alternatively, I would be fine if the _interval_ was fixed, like it runs once when the process starts and then once per day. Each time it runs it deletes orders older than `CleanupTTL`.

Reply:
I was thinking this too. A fixed interval of once per day sounds good to me